Fixes #12260 - tests updated to work with Foreman 1.10 API

mbacovsky opened the pull request:
- API description with examples was updated to match API from Foreman 1.10
- id_resolver_test was updated to test on arbitrary API description
- tests updated 
 
Questions to answer:
Q: Do we want to support all API v2 variants or should we stick to specific Foreman version(s)?
A: Master should support two latest stable foreman releases + foreman develop

Q: Should we better test id_resolver on generic data?
A: Yes. Fixed.

tstrachota commented:
"Alle Faktwerte auflisten" did you intentionally create translated apidoc dump?:) :de:

mbacovsky commented:
Thank you for your question. I took the apidoc dump from the issue attachment. So your question should be better rephrased "Did I intentionally create translated apidoc dump?". Did you? :poodle:

mbacovsky commented:
README about updating test data added.

tstrachota commented on lib/hammer_cli_foreman/id_resolver.rb:
```diff
@@ -174,7 +174,6 @@ def resolved_call(resource_name, action_name, options)
         search_options[param.name] ||= send(param.name, scoped_options(param.name.gsub(/_id$/, ""), options))
       end
       search_options = route_options(options, action).merge(search_options)
-
```
I'd rather avoid whitespace changes if it's the only modification in the file.

tstrachota commented:
Let's keep the data in english, please https://github.com/mbacovsky/hammer-cli-foreman/pull/1

tstrachota commented on test/unit/data/README.md:
````diff
@@ -0,0 +1,22 @@
+ 1. update the following line in `test/unit/test_helper.rb` to match the new default Foreman version
+
+    ```ruby
+    FOREMAN_VERSION = Gem::Version.new(ENV['TEST_API_VERSION'] || '1.10')
````
Can you please document the usage of TEST_API_VERSION somewhere? The section about testing in the main readme is probably a good candidate.

tstrachota commented on test/unit/dependency_resolver_test.rb:
```diff
@@ -9,15 +9,24 @@
-        :compute_resources, :organizations, :locations
-      ].sort_by{|sym| sym.to_s}
+      resources = resolver.resource_dependencies(resource).map(&:name).sort_by{ |sym| sym.to_s }
+      if FOREMAN_VERSION < Gem::Version.new('1.10')
```
Since you added `test/unit/data/test_api.json` I propose to switch this test to use the fake data too. What do you think?

tstrachota commented on test/unit/option_builders_test.rb:
```diff
@@ -123,9 +123,13 @@ def resource_names(builders)
-        :compute_resources
-      ]
+      resources = resource_names(builders_by_class(HammerCLIForeman::DependentSearchablesOptionBuilder)).sort
+      if FOREMAN_VERSION < Gem::Version.new('1.10')
```
Same as in https://github.com/theforeman/hammer-cli-foreman/pull/203/files#r46411880

tstrachota commented:
Does it make sense to have the tests fenced with `version < 1.10` when you removed all the older apidoc dumps? I'd either add dumps for 1.9 or remove the conditions.

mbacovsky commented:
@tstrachota in the update I've addressed your concerns. It is in separate commit for easier review. I can squash when it is acked. I've converted option_builder_test to use test_api, but there are still the taxonomies from foreman_api. It is not ideal, but I'll have to extend the test_api to mimic the foreman_api first.

tstrachota commented on test/unit/host_test.rb:
```diff
@@ -73,7 +73,11 @@
         it "should output status" do
           cmd.stubs(:context).returns(ctx.update(:adapter => :test))
-          proc { cmd.run(with_params) }.must_output "#Status#Power#\n#missing#running#\n"
+          if FOREMAN_VERSION < Gem::Version.new('1.10')
```
Was this one left unchanged intentionally?

mbacovsky commented:
@tstrachota PR was rebased, remaining concerns were fixed. I fixed the option builder test and replaced all the resources with the ones from the fake API.  Do not forget to squash before merge or let me know and I'll squash for you.

mbacovsky commented:
Squashed

tstrachota commented:
LGTM, thanks @mbacovsky. Good job with changing the tests to use the fake apidoc!
:dolphin:
